contrib/babelfishpg_tsql/src/fts_parser.y

```diff
@@ -151,21 +160,37 @@ char* translate_simple_term(const char* inputStr) {
         while (*inputPtr != '\0') {
             if (*inputPtr == ' ') {
                 // Replace space with "<->"
+                while (*(inputPtr + 1) == ' ') {
+                    // Handle multiples spaces between words and skip over additional spaces
+                    inputPtr++;
+                }
+                if (outputPtr - output + 3 > outputSize) {
+                    // Output buffer overflow
+                    pfree(trimmedInputStr);
+                    pfree(output);
+                    return NULL;
+                }
                 *outputPtr++ = '<';
                 *outputPtr++ = '-';
                 *outputPtr++ = '>';
             } else {
                 // Copy the character
+                if (outputPtr - output + 1 > outputSize) {
+                    // Output buffer overflow
+                    pfree(trimmedInputStr);
+                    pfree(output);
+                    return NULL;
+                }
                 *outputPtr++ = *inputPtr;
```

Comment:
When do you copy the null terminator over? Right now as far as I can tell you never set it, and just rely that the palloc() call puts a null terminator at the end, which is not correct.